src/hooks/useLoginProfile.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  CodeChallengeMethod,
+  exchangeCodeAsync,
+  makeRedirectUri,
+  refreshAsync,
+  ResponseType,
+  useAuthRequest
+} from 'expo-auth-session';
+import { useEffect, useState } from 'react';
+
+import * as appJson from '../../app.json';
+import { secrets } from '../config';
+import { addToStore, readFromStore } from '../helpers';
+
+const PROFILE_ACCESS_TOKEN = 'profileAccessToken';
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+const profile = secrets[namespace]?.profile;
+const serverUrl = profile?.serverUrl;
+
+const keycloakDiscovery = {
+  authorizationEndpoint: serverUrl + '/auth',
+  tokenEndpoint: serverUrl + '/token',
+  revocationEndpoint: serverUrl + '/revoke'
+};
+
+const CLIENT_ID = profile?.clientId;
+const CLIENT_SECRET = profile?.clientSecret;
+const SCOPES = ['openid', 'email', 'profile', 'phone', 'address'];
+
+export const useLoginProfile = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const redirectUri = makeRedirectUri({
+    scheme: appJson.expo.scheme,
+    path: 'redirect'
+  });
+
+  const [request, response, promptAsync] = useAuthRequest(
+    {
+      clientId: CLIENT_ID,
+      codeChallengeMethod: CodeChallengeMethod.S256,
+      redirectUri,
+      responseType: ResponseType.Code,
```

Comment:
why do we add it here, if it is the default? 

![image](https://github.com/user-attachments/assets/fd34d716-f218-4273-924f-fd27b2eb3cd8)

https://docs.expo.dev/versions/latest/sdk/auth-session/#authrequestconfig